### DISCUSSION TITLE
Do the collaborators of a private repo need a paid github plan too?

### DISCUSSION THREAD

**Discussion by anantoghosh** [community/community#22051] [New to GitHub]
<p>Do the collaborators of a private repo need a paid github plan too?</p>
<p>For example - If I get the developer paid plan, then create a private repo, would I be able to add a person to the repo who only has a free github account?</p>

**Comment by lee-dohm** [comment, marked as answer]
<blockquote>
<p><a class="mention" href="/anantoghosh">@anantoghosh</a> wrote:</p>
</blockquote>
<blockquote>
<p>Do the collaborators of a private repo need a paid github plan too?</p>
</blockquote>
<p>No. Users who are added as collaborators to another user’s or organization’s private repositories can be on the free plan.</p>

**Reply by Hakkimhong** [reply]
Code money gram 100 USD

**Comment by anantoghosh** [comment]
<p>Thanks :smiley:</p>

**Comment by timexRiel** [comment]
<p>From this conversation, I figured that outside collaborator doesn’t need to be on a paid plan to have access to an org’s private repo, right? However, will this not account to the org’s paid seat count?</p>
<p>I asked because it looks like an outside collaborator of private repos takes up paid seats of the organization, based on the following links.</p>
<p><a href="https://help.github.com/articles/removing-paid-seats-from-your-organization/" rel="nofollow noopener">https://help.github.com/articles/adding-outside-collaborators-to-repositories-in-your-organization/</a></p>
<p>and</p>
<p><a href="https://help.github.com/articles/removing-an-outside-collaborator-from-an-organization-repository/" rel="nofollow noopener">https://help.github.com/articles/removing-an-outside-collaborator-from-an-organization-repository/</a></p>
<p>Thanks!</p>

**Comment by solocommand** [comment]
<p>Correct, adding an outside collaborator to a repository will take up a seat in a paid Github license.</p>

**Reply by rbaduwal** [reply]
Is there any other way to accomplish it?

**Reply by solocommand** [reply]
I believe this has changed over the years. My understanding is that an organization or user on a *free* plan can add a collaborator to a private repository without incurring additional cost.

If the organization has a *paid* plan, the external collaborator will still take up a paid seat.

**Comment by jveitchmichaelis** [comment]
<p>Note this only applies to personal private repositories. If you have an organisation and you want to add an external collaborator to a private repository, the organisation has to have a seat available (which is not free).</p>
<p>It’s a bit irritating, but it’s presumably to stop the problem of organisations having a single paid user and a ton of free collaborators without paying for more per-user licenses.</p>

**Comment by shuangg** [comment]
<p>&gt;&gt; it’s presumably to stop the problem of organisations having a single paid user and a ton of free collaborators without paying for more per-user licenses.</p>
<p>How could this stop the problem? It seems to only encourgae it.</p>

**Comment by madalinignisca** [comment]
<p>I just discovered also, I need to pay extra seats in my organization for a few extra collaborators.</p>
<p>Let me get the case:</p>
<p>I have an organization and I’d like for my internal people to be on paid seats.</p>
<p>I’d like also to be able to have for some side projects we do for a some extra money, to be able to add a few collaborators to the private repos. As this are like probably 1 or 2 projects ocassionally, mostly for 1-3 months, the collaborators would be 1-2 people from the companies we’d be doing this projects, 99% non-developers, so they can see some activity and be able to collaborate on the issue/project board. Possible we’d not remove them, as the issue board should be there for reporting bugs.</p>
<p>But, Github enforces us to have them paid, although they are really outside the real usage we do internally.</p>
<p>There should be a difference for 2 type of collaborators. One that is like a full developer, mostly at same level like internal people, and to be paying a seat for, and another type, like limited readonly repo, participate in issues only type that could be free.</p>
<p>Attlasian/Jira offers this, with their support feature, where I can add free collaborators with only one thing - report issues.</p>
<p>We are a team of 3. With 3-4 small extra projects added each year, I’d endup with 6-12 extra seats to pay monthly. It is also embarrassing to tell the clients I need to kick them off.</p>
<p>This leads me to the critical need to create many organizations, or bring them on my pro account as free collaborators - not looking too professional.</p>
<p>So I guess the prices are decided by some non-developers hungry for money, like lots of people where in M$ decades ago, and the CEO of M$ should come to them and smash the same great current philosophy M$ has for a lot of products. This will keep us from migrating to other more financially friendly platforms.</p>

**Reply by claesenm** [reply]
We are in the exact same situation.

**Comment by Wazbat** [comment]
<p>Apologies for bumping this old thread, but I have a similar issue to <a class="mention" href="/madalinignisca">@madalinignisca</a></p>
<p>If anything this makes it hard to justify paying for Github, as once we bought a few seats that now means we can no longer add outside collaborators. For example, we inherited a project and brought the code in house, but now this means we are forced to remove the previous developers from the repository. They don’t make contributions, but they need read access to the code as part of the support contract. If we’re on the free plan, we’d be able to add them.</p>
<p>There’s other cases too, like for example I’m suggest migrating from Trello to Github’s project boards. However that means that we need to pay for a seat for every person who wants to access even just the project boards which is a bit ridiculous. There should be some kind of access for users that only want to see the boards, or maybe create issues in a repository. This makes it really hard to sell anyone on moving everything we do to Github</p>
<p>This is a bit ridiculous. We actually get more usability out of being on the free plan</p>

**Comment by codeagencybe** [comment]
Pff this problem is still existing after such a long time. 
It's basically impossible to have external partners onboard on private repo's without breaking your (org) wallet. 

What's even more stupid, is that many of those external partners (at least in our case) ALREADY PAY for a Github pro account on their end. So if I invite them into my private repo to help out with code, like a temporary freelance project, you end up sucking up all the costs for those freelancers as well.
As an org, you actually pay again for the same feature sets for those that already pay it personally. It doesn't resonate at all. 

Why is there no clear price diff for team members vs external collabs? And also an option to choose to pass the bill to freelancer. 
Then I can choose if I want to bear the cost of onboarding the collaborator in my org or if they have to do it on their own. 
When they click the invite they are offered with some options:
Already on a pro or higher plan -> no issue, can join instantly
Not yet on a pro or higher plan -> upsell to upgrade page on their own account

**Comment by dhrp** [comment]
I came here too; to understand how to add business people for seeing and making issues; but at the same time the current pricing of $4 per user seems very reasonable to me for what you get in return. Even if you pay for a couple of extra users.

**Reply by codeagencybe** [reply]
That is not the problem people are addressing here. 

The issue is: 
Let's assume you have 20 external collaborators to work with, and each of them individually already pay for their own Github pro plan. 
Now you invite them into your organisation private repo, Github will charge you another 20 seats into your company billing account. 

If I do exactly the same with a PERSONAL private or public repo, there is not a problem, not a single charge at all. 
I can invite as many collaborators I want, zero upsell or charge happening. Even if all those collaborators have free Github accounts. 

If a personal repo, private or public can have unlimited free collaborators, then why can't we have it for organisation repo? 
Especially if each developer already paid himself for his own personal pro account. 
Github is now basically charging me DOUBLE for each collaborator account. 

I can understand they charge me for onboarding team members from my own company. But in that case, that is also already solved as each team member has a paid seat anyway from my company. That's just how any SaaS works, you pay for each company team member. 

My problem is the double agenda they are doing and charging companies again for something the collaborator outside the company already paid individually.

**Reply by claesenm** [reply]
Not everyone is on a $4 license though. We get to pay full enterprise license for each external collaborator that pulls a repo once every full moon. This is not justified in terms of costs.

**Reply by codeagencybe** [reply]
> Not everyone is on a $4 license though. We get to pay full enterprise license for each external collaborator that pulls a repo once every full moon. This is not justified in terms of costs.

What bothers me more is, even if that person you invite is already paying for a PRO plan on their own behalf, as an org you have to pay AGAIN the same seat. Github is just eating double margin for the same person that is being invited into your org repo. Wether it costs 4$ or not is irrelevant, the problem is the double cost/charge in the first place.

**Comment by rochelledalziel** [comment]
Commenting in support of others who are querying this potential double-charge (or greater) business model that Github have going. We are an organisation of 22/23 core internal members, but we are paying for 61 seats due to the number of outside collaborators we have. It's getting a bit out of hand and is a large drain on our cashflow which I don't think we should need to shoulder (however I'm not a tech/dev person so I don't really understand how the system works).
Surely outside collaborators should be free or a much lower rate than a fully-paid member. 
And why is it not possible to export the list of outside collaborators?! Such a nuisance :(

**Reply by codeagencybe** [reply]
Yes, you are right. 

Imho, if an "outside" member/collaborator is already paying for their own Github pro account, I don't understand why my company needs to pay AGAIN for giving the same features... 

There is a disconnect in how Github works. 
If you create a private repo in **_your personal account_**, and then invite external people, you are not charged. No matter if the collaborator has a free or paid plan on their own. 
if you create a private repo in **_your organisation account_**, and then invite external people, Github charges them as extra seats to your organisation, **_even if the external collaborator already have their own paid plan_**. 

I don't understand why this designed like this. This is just double charging the same thing. 

I just ended up creating repo's under my personal account because of this stupid rule. If you work very often with external collaborators, this cost just explodes for orgs. But mostly it's completely unfair that I have to pay again for every seat that already has a pro plan. 

I can accept that GH requires at minimum every collaborator must be on a paid plan and enforce this but they should let companies choose if they want to pass on the bill to the external collaborator. And if they already are on a paid plan, then the seat should be free. 

This is just money grabbing at current design.

**Reply by themightychris** [reply]
This is such a barrier to collaboration. I'd use GitHub projects a lot more if we could add people from partner companies to private projects without them taking up seats in our org when they're ALREADY COVERED by seats in their own org

**Comment by Siva2231** [comment]
i want private repo for one of my project and also need as per below instruction only 
contact me on Whatsapp - 6379316208 or Email - sivaprasad22062000@gmail.com
I will provide you 300 USD dollar if its meet my requirement!!

Repositories must be private on GitHub
Submit only repositories that are private on GitHub (or not on GitHub at all). The platform automatically probes public GitHub for your repo and rejects any match — including forks, mirrors, and renamed copies. The same applies to other public hosts (GitLab, Bitbucket, public Gitea instances, etc.): if your code is reachable publicly anywhere on the internet, don’t submit it.

Submitting a public or open-source repository will be flagged, will result in rejection, and may result in removal from the project.
You must own the IP
By submitting, you confirm you own the code outright. That means it’s code you personally authored — typically a personal project, a fork you’ve contributed substantially to, or solo work you have full rights to. Do not submit code from your employer (current or former), even with written permission. Do not submit proprietary code, code under restrictive licenses you can’t grant rights to, or code that contains third-party IP. Submissions are conditionally assigned to AfterQuery on approval (per the Repository Submission Agreement), so the IP must be yours to assign.

Quality matters
We’re looking for real, meaningful codebases that reflect substantive work. Personal projects are fine — including solo work — as long as they’re significant. Good submissions include long-running personal projects you’ve invested time in, side projects with genuine scope and history, or multi-author collaborations.

What we don’t accept: homework, course assignments, hackathon submissions, tutorial follow-alongs, boilerplate starters or scaffolds, throwaway experiments, and fresh-init repos. Repositories that are too small, too recent, too single-purpose, or that obviously read as a school project, a weekend hack, or scaffolding lifted from a starter template will be rejected — both by automated gates and by the human reviewer who looks at every submission.

The platform also checks for duplicates and near-duplicates. If your repository is too similar to one that’s already in the system (including forks and renamed copies), it will be rejected.

Supported languages
TypeScript, JavaScript, Java, Rust, Go, and Python are fully supported with pre-built templates. You can also select “Other” and specify the language manually — you’ll just need to write the test runner and parser scripts yourself.
